### ISSUE TITLE
state / stateChange behavior

### ISSUE THREAD

**Issue by Cody-G-G** [sandeepmistry/noble#513]
Disclaimer: I am actually using https://github.com/jacobrosenthal/react-native-ble with Android, however I am looking to find out whether I simply have a poor understanding of how this works, or if this is an issue with either react-native-ble or noble.

I am a bit confused as to how state / stateChange work...I initially assumed that a stateChange event would be automatically emitted when the user enables / disables bluetooth, but this doesn't seem to be the case. I then thought I should check whether bluetooth is on or off (whenever I need it), and change the state + emit event myself (" noble.state = 'poweredOn; noble.emit('stateChange', 'poweredOn' "), but even though this changes the noble.state to whatever I want it to, when I try and scan for devices, the scan actually starts but doesn't actually find anything (even though there are devices to be found).

In short: if I start the app with bluetooth off, I don't seem to be able to find devices even though I subsequently enable bluetooth, change noble state to 'poweredOn', and emit the appropriate 'stateChange' event.

I am sure I'm missing something obvious or grossly misunderstanding how these should be used, but I've been trying to figure it out with no success. I would greatly appreciate anyone shedding some light on this, and whether this behavior is normal (but I'm doing it wrong), or is it a bug?

Also, I was wondering whether noble supports serial data by any chance? After having a look around I believe the answer is no, but wanted to make sure? If no, is there a plan to implement this functionality at some point?

**Comment by sandeepmistry** [comment]
> I am a bit confused as to how state / stateChange work...I initially assumed that a stateChange event would be automatically emitted when the user enables / disables bluetooth, but this doesn't seem to be the case.

That's the expected behaviour.

> Also, I was wondering whether noble supports serial data by any chance? After having a look around I believe the answer is no, but wanted to make sure? If no, is there a plan to implement this functionality at some point?

There is no standard for serial data over BLE. Even if there was, noble would only be used as the lower level to build upon by another Node.js library/module.

Closing for now, I suggest you follow up with the `react-native-ble` maintainers (and observe some patience).
